FLT/Mathlib/Topology/Algebra/Module/ModuleTopology.lean

```diff
@@ -470,6 +472,288 @@ instance instProd' : IsModuleTopology (A × B) (M × N) := inferInstance
 
 end Prod
 
+section ModuleProd
+
+/- This should probably replace the Prod section above ... reduces CommRing → Ring. -/
+
+variable {R S M N : Type*} [Semiring R] [Semiring S] [TopologicalSpace R] [TopologicalSpace S]
+  [AddCommMonoid M] [AddCommMonoid N] [Module R M] [Module S N]
+
+/-- The R × S module structure on M × N. -/
+local instance : Module (R × S) (M × N) := ModuleProd.instModuleProd
+
+/-- Inclusion `M → M × N` by `a ↦ (a, 0)`. -/
+abbrev incl1 : M → M × N :=
+  fun a => (a, 0)
+
+/-- Inclusion `N → M × N` by `a ↦ (0, a)`. -/
+abbrev incl2 : N → M × N :=
+  fun b => (0 , b)
+
+lemma induced_continuous_add : @ContinuousAdd M
+    (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N))) _ := by
+  suffices h : @Continuous (M × M) (M × N) (@instTopologicalSpaceProd M M
+      (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+      (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N))))
+      ((moduleTopology (R × S) (M × N))) (fun (a : M × M) => (a.1 + a.2, (0 : N))) by
+    convert (@Topology.IsInducing.continuous_iff (M × M) M (M × N) (fun (a : M × M) ↦ a.1 + a.2)
+        (incl1) (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+        (@instTopologicalSpaceProd M M
+        (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+        (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N))))
+        (moduleTopology (R × S) (M × N)) _).mpr h
+    · constructor <;> intro h
+      · exact ContinuousAdd.continuous_add
+      · rw [continuous_def] at h
+        use h
+    · rw [@Topology.isInducing_iff]
+  have h : (fun (a : M × M) => (a.1 + a.2, (0 : N))) =
+      (fun (b : (M × N) × (M × N)) => (b.1.1 + b.2.1, b.1.2 + b.2.2)) ∘
+      (fun (a : M × M ) => (((a.1, 0) : (M × N)) , (((a.2, 0)) : (M × N)))) := by
+    ext a
+    all_goals simp
+  rw [h]
+  refine @Continuous.comp (M × M) ((M × N) × M × N) (M × N) (@instTopologicalSpaceProd M M
+    (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+    (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N))))
+    (@instTopologicalSpaceProd (M × N) (M × N) (moduleTopology (R × S) (M × N))
+    (moduleTopology (R × S) (M × N))) (moduleTopology (R × S) (M × N))
+    (f := (fun (a : M × M ) => (((a.1, 0) : (M × N)) , (((a.2, 0)) : (M × N)))))
+    (g := (fun (b : (M × N) × (M × N)) => (b.1.1 + b.2.1, b.1.2 + b.2.2))) ?_ ?_
+  · convert ContinuousAdd.continuous_add
+    exact ModuleTopology.continuousAdd (R × S) (M × N)
+  · refine @Continuous.prodMap (M × N) (M × N) M M (moduleTopology (R × S) (M × N))
+      (moduleTopology (R × S) (M × N))
+      (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+      (TopologicalSpace.induced (incl1) (moduleTopology (R × S) (M × N)))
+      (f := incl1) (g := incl1) ?_ ?_
+    all_goals exact continuous_iff_le_induced.mpr fun U a ↦ a
```

Comment:
After this heroic effort you can deduce this much cleaner statement:
```
lemma induced_continuous_add' [TopologicalSpace M] [TopologicalSpace (M × N)]
    [IsModuleTopology (R × S) (M × N)] (hM : Topology.IsInducing (AddMonoidHom.inl M N)) :
    ContinuousAdd M := by
  obtain ⟨rfl⟩ := hM
  have := eq_moduleTopology (R × S) (M × N)
  convert induced_continuous_add
```
Is this cleaner lemma what you should be proving in the first place? Not sure. If I could understand what you were doing then I might have a stab at proving this directly, but what I'm unclear about is whether it's enough for you because at this point I'm not entirely sure why you need these results.